lib/glific/jobs/user_jobs.ex

```diff
@@ -0,0 +1,102 @@
+defmodule Glific.Jobs.UserJob do
+  @moduledoc false
+  use Ecto.Schema
+  import Ecto.Changeset
+  import Ecto.Query, warn: false
+  alias __MODULE__
+
+  alias Glific.{
+    Partners.Organization,
+    Repo
+  }
+
+  @required_fields [
+    :status,
+    :type,
+    :total_tasks,
+    :tasks_done,
+    :organization_id,
+    :all_tasks_created
+  ]
+  @optional_fields [:errors]
+
+  @type t() :: %__MODULE__{
+          __meta__: Ecto.Schema.Metadata.t(),
+          id: non_neg_integer | nil,
+          status: String.t() | nil,
+          type: String.t() | nil,
+          total_tasks: non_neg_integer | nil,
+          tasks_done: non_neg_integer | nil,
+          organization_id: non_neg_integer | nil,
+          errors: map() | nil,
+          all_tasks_created: boolean | false,
+          organization: Organization.t() | Ecto.Association.NotLoaded.t() | nil,
+          inserted_at: :utc_datetime | nil,
+          updated_at: :utc_datetime | nil
+        }
+
+  schema "user_jobs" do
+    field(:status, :string, default: "pending")
+    field(:type, :string)
+    field(:total_tasks, :integer)
+    field(:tasks_done, :integer)
+    field(:errors, :map, default: %{})
+    field(:all_tasks_created, :boolean, default: false)
+
+    belongs_to(:organization, Organization, foreign_key: :organization_id)
+
+    timestamps(type: :utc_datetime)
+  end
+
+  @doc """
+  Standard changeset pattern we use for all data types
+  """
+  @spec changeset(UserJob.t(), map()) :: Ecto.Changeset.t()
+  def changeset(user_job, attrs) do
+    user_job
+    |> cast(attrs, @required_fields ++ @optional_fields)
+    |> validate_required(@required_fields)
+    |> foreign_key_constraint(:organization_id)
+  end
+
+  @spec create_user_job(map()) :: %UserJob{}
+  def create_user_job(attrs) do
+    %UserJob{}
+    |> UserJob.changeset(attrs)
+    |> Repo.insert!()
+  end
+
+  @doc """
+  Updates a user job with the given changeset.
+  """
+  def update_user_job(user_job, changes) do
+    user_job
+    |> changeset(changes)
+    |> Repo.update()
+  end
+
+  @spec list_user_jobs(map()) :: [UserJob.t()]
+  def list_user_jobs(args \\ %{}) do
+    args
+    |> Glific.add_limit()
+    |> Repo.list_filter_query(UserJob, &Repo.opts_with_name/2, &filter_with/2)
```

Comment:
This query doesn't give what we expect, the actual query it return is 
`#Ecto.Query<from u0 in Glific.Jobs.UserJob, limit: ^25>`

This is because we have to change this ` args = %{status: "pending", all_tasks_created: true}` to ` args = %{filter: %{status: "pending", all_tasks_created: true}}`. Basically filter function expects a filter key